src/lib/markbind/src/parser.js

```diff
@@ -219,6 +219,25 @@ Parser.prototype._preprocess = function (node, context, config) {
       const $ = cheerio.load(segmentSrc);
       const htmlContent = $(includeSrc.hash).html();
       let actualContent = htmlContent;
+
+      if (actualContent === null) {
+        if (isOptional) {
+          // set empty content for optional segment include that does not exist
+          actualContent = '';
+        } else {
+          const error = new Error(
+            `No such segment ${includeSrc.hash.substring(1)}`,
```

Comment:
1. The included file should be mentioned to make it clearer.
2. The file referencing the included file should also be mentioned, because the error is printed on the console as well.

So the final error message should be something like this:
```javascript
          const error = new Error(
            `No such segment '${includeSrc.hash.substring(1)}' in file: ${actualFilePath}`
              + `\nMissing reference in ${element.attribs[ATTRIB_CWF]}`,
          );
```

Reply:
Good point, I have changed the error message to be more detailed!